modules/top-level/test.nix

```diff
@@ -1,10 +1,90 @@
-{ lib, ... }:
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.test;
+in
 {
   options.test = {
+    name = lib.mkOption {
+      type = lib.types.str;
+      default = "nixvim-check";
+      description = "The test derivation's name.";
+    };
+
     runNvim = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to run `nvim` in the test.";
       default = true;
     };
+
+    # Output
+    test = lib.mkOption {
```

Comment:
I'm not a big fan of naming this `config.test.test`...

Reply:
test.derivation?